optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingImporter.java

```diff
@@ -0,0 +1,511 @@
+                .withContentTagSet(new HashSet<>())
+                .withPreferredRoomTagSet(getPreferredRoomTagSetForTalkOfType(talkTypeName))
+                .withPreferredTimeslotTagSet(new HashSet<>())
+                .withProhibitedRoomTagSet(getProhibitedRoomTagSetForTalkOfType(talkTypeName))
+                .withProhibitedTimeslotTagSet(new HashSet<>())
+                .withRequiredRoomTagSet(new HashSet<>())
+                .withRequiredTimeslotTagSet(new HashSet<>())
+                .withSectorTagSet(new HashSet<>())
+                .withUndesiredRoomTagSet(new HashSet<>())
+                .withUndesiredTimeslotTagSet(new HashSet<>());
+        return talk;
+    }
+
+    private Set<String> getPreferredRoomTagSetForTalkOfType(String talkTypeName) {
+        Set<String> preferredRoomTagSet = new HashSet<>();
+        if (Arrays.asList(smallRoomsTypeNames).contains(talkTypeName)) {
+            preferredRoomTagSet.add(SMALL_ROOM_TAG);
+        } else if (Arrays.asList(mediumRoomsTypeNames).contains(talkTypeName)) {
+            preferredRoomTagSet.add(MEDIUM_ROOM_TAG);
+        } else {
+            preferredRoomTagSet.add(LARGE_ROOM_TAG);
+        }
+        return preferredRoomTagSet;
+    }
+
+    private Set<String> getProhibitedRoomTagSetForTalkOfType(String talkTypeName) {
+        Set<String> prohibitedRoomTagSet = new HashSet<>();
+        if (Arrays.asList(largeRoomsTypeNames).contains(talkTypeName)) {
+            prohibitedRoomTagSet.add(SMALL_ROOM_TAG);
+            prohibitedRoomTagSet.add(MEDIUM_ROOM_TAG);
+        } else if (Arrays.asList(mediumRoomsTypeNames).contains(talkTypeName)) {
+            prohibitedRoomTagSet.add(SMALL_ROOM_TAG);
+        }
+
+        return prohibitedRoomTagSet;
+    }
+
+    private Set<TalkType> getTalkTypeSetForCapacity(int capacity) {
+        Set<TalkType> talkTypeSet = new HashSet<>();
+        List<String> typeNames = new ArrayList<>();
+        if (capacity < 100) {
+            typeNames.addAll(Arrays.asList(smallRoomsTypeNames));
+        } else if (capacity < 1000) {
+            typeNames.addAll(Arrays.asList(mediumRoomsTypeNames));
+            typeNames.addAll(Arrays.asList(smallRoomsTypeNames));
+        } else {
+            typeNames.addAll(Arrays.asList(largeRoomsTypeNames));
+            typeNames.addAll(Arrays.asList(mediumRoomsTypeNames));
+            typeNames.addAll(Arrays.asList(smallRoomsTypeNames));
+        }
+
+        for (String talkTypeName : typeNames) {
+            TalkType talkType = talkTypeNameToTalkTypeMap.get(talkTypeName);
+            if (talkType != null) {
+                talkTypeSet.add(talkType);
+            }
+        }
+
+        return talkTypeSet;
+    }
+
+    private JsonObject readJsonObject(String url) {
+        try (InputStream inputStream = openConnectionCheckRedirects(new URL(url).openConnection())) {
+            JsonReader jsonReader = Json.createReader(inputStream);
+            return jsonReader.readObject();
```

Comment:
Soft suggestion: this line and the return type are the only differences between this method and `readJsonArray`. Maybe passing there  lambda and using type parameter would avoid duplication.